Sources/Controls/UIBarButtonItem+Combine.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UIBarButtonItem+Combine.swift
+//  CombineCocoa
+//
+//  Created by Shai Mishali on 12/08/2019.
+//  Copyright © 2019 Shai Mishali. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+public extension UIBarButtonItem {
+  /// A publisher which emits whenever this UIBarButtonItem is tapped.
+  var tapPublisher: AnyPublisher<Void, Never> {
+    return TapPublisher(barButtonItem: self).eraseToAnyPublisher()
+  }
+}
+
+private extension UIBarButtonItem {
+  struct TapPublisher: Publisher {
+    public typealias Output = Void
+    public typealias Failure = Never
+
+    weak private var barButtonItem: UIBarButtonItem?
```

Comment:
Why `weak`, `var` and optional? 🤔

I think `private let barButtonItem: UIBarButtonItem` should be fine.

Reply:
My guess is so that when the bar button goes away we send a completed for any subscriber that attaches to this publisher.

Reply:
private let will hold a strong reference to the UIBarButtonItem. I took the thoughts from RxCocoa's implementation.

Reply:
Sorry, but I don't understand the problem of having a strong reference to the `UIBarButtonItem`. We don't have a strong reference cycle somewhere.

Reply:
I was following Kruno's work here: https://github.com/ReactiveX/RxSwift/blob/master/RxCocoa/iOS/UIBarButtonItem+Rx.swift#L55

I think the reasoning is really the "dispose" part so I've altered my impl a bit. Take a second look please :)